lib/jsdom/living/events/EventTarget-impl.js

```diff
@@ -255,6 +241,48 @@ function invokeEventListeners(listeners, target, eventImpl) {
 
 const wrappedListener = Symbol("inline event listener wrapper");
 
+/**
+ * Normalize the event listeners options argument in order to get always a valid options object
+ * @param   {Object} options         - user defined options
+ * @param   {Array} defaultBoolKeys  - boolean properties that should belong to the options object
+ * @param   {Array} keysBlackList    - options not allowed (optional)
+ * @returns {Object} object containing at least the  "defaultBoolKeys"
+ */
+function normalizeEventHandlerOptions(options, defaultBoolKeys, keysBlackList) {
+  // normalize the options argument in order to get always an object
+  if (options === null || options === undefined) {
+    options = {};
+  } else if (typeof options !== "object") {
+    options = { capture: options };
+  }
+
+  let defaultOptions;
+  let additionalOptions;
+  keysBlackList = keysBlackList || [];
+
+  try {
+    // add the default options properties
+    defaultOptions = defaultBoolKeys.reduce((obj, key) => {
+      obj[key] = Boolean(options[key]);
+      return obj;
+    }, {});
+
+    // somehow according to the specs users could pass also other properties as options
+    additionalOptions = Object.keys(options).filter(k => !defaultBoolKeys.includes(k)).reduce((obj, key) => {
+      if (!keysBlackList.includes(key)) {
+        obj[key] = options[key];
+      }
+
+      return obj;
+    }, {});
+  } catch (e) {
+    // should the errors be simply swallowed here?!
```

Comment:
Errors should be rethrown, not caught---otherwise test failures, like the assert_unreached, will not surface correctly!